modules/local/circos/circos_prepare.nf

```diff
@@ -0,0 +1,37 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from '../functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process CIRCOS_PREPARE {
+    tag "$meta.id"
+    label 'process_medium'
+    label 'error_ignore'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:'') }
+
+    conda (params.enable_conda ? "bioconda::bioconductor-rtracklayer=1.50.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bioconductor-rtracklayer:1.50.0--r40h7f5ccec_2"
+    } else {
+        container "quay.io/biocontainers/bioconductor-rtracklayer:1.50.0--r40h7f5ccec_2"
+    }
+
+    input:
+    tuple val(meta), path(bedpe), val(ucscname), path(gtf), path(chromsize)
+
+    output:
+    tuple val(meta), path("circos/*")               , emit: circos
+    path "*.version.txt"                            , emit: version
+
+    script:
+    """
+    circos.r \\
```

Comment:
yes, they will be used always together. I separate them to make sure the container will be ready for each step. And they will be connected in `subworkflow`.

Reply:
I'm not sure I understand. The container is always prepared before running the process anyway. 
Also it's better practice to bundle short tasks into a larger task, particularly if they're related.

Reply:
I agree. I need some time to figure out how to make it possible for the dependencies available in one container.

Reply:
Oh, I thought it was already part of the same container/tool package. It's been a long time since I used circos. One possibility might be a mulled container, but if they're separate tools then one should keep them separate.